File: tools/monitoring/test-aleo-connection.js
#!/usr/bin/env node

// Test Aleo Connection
// Verifies we can connect to Aleo API and fetch contract data

const https = require('https');

const contracts = [
    'hallucination_verifier.aleo',
    'agent_registry_v2.aleo',
    'trust_verifier_v2.aleo'
];

const endpoints = [
    'https://api.explorer.provable.com/v1',
    'https://api.explorer.aleo.org/v1',
    'https://api.explorer.aleo.org/v1/testnet3'
];

async function testEndpoint(endpoint, contractId) {
    return new Promise((resolve) => {
        const url = `${endpoint}/testnet3/program/${contractId}`;
        console.log(`\nTesting: ${url}`);
        
        https.get(url, {
            headers: {
                'Accept': 'application/json',
                'User-Agent': 'Lamassu-Labs-Test/1.0'
            }
        }, (res) => {
            let data = '';
            
            res.on('data', (chunk) => {
                data += chunk;
            });
            
            res.on('end', () => {
                if (res.statusCode === 200) {
                    try {
                        const json = JSON.parse(data);
                        console.log(`✅ Success! Found program: ${contractId}`);
                        console.log(`   Status Code: ${res.statusCode}`);
                        if (json.id) console.log(`   Program ID: ${json.id}`);
                        if (json.mappings) console.log(`   Mappings: ${json.mappings.length}`);
                        if (json.functions) console.log(`   Functions: ${json.functions.length}`);
                        resolve({ success: true, endpoint, contractId, data: json });
                    } catch (e) {
                        console.log(`❌ Failed: Invalid JSON response`);
                        resolve({ success: false, endpoint, contractId, error: 'Invalid JSON' });
                    }
                } else {
                    console.log(`❌ Failed: HTTP ${res.statusCode}`);
                    console.log(`   Response: ${data.substring(0, 100)}...`);
                    resolve({ success: false, endpoint, contractId, statusCode: res.statusCode });
                }
            });
        }).on('error', (err) => {
            console.log(`❌ Failed: ${err.message}`);
            resolve({ success: false, endpoint, contractId, error: err.message });
        });
    });
}

async function testTransactions(endpoint, contractId) {
    const paths = [
        `/testnet3/program/${contractId}/transitions?limit=5`,
        `/testnet3/transitions?program=${contractId}&limit=5`,
        `/testnet3/transactions?program=${contractId}&limit=5`
    ];
    
    for (const path of paths) {
        const url = endpoint + path;
        console.log(`\nTesting transactions: ${url}`);
        
        const result = await new Promise((resolve) => {
            https.get(url, {
                headers: {
                    'Accept': 'application/json',
                    'User-Agent': 'Lamassu-Labs-Test/1.0'
                }
            }, (res) => {
                let data = '';
                
                res.on('data', (chunk) => {
                    data += chunk;
                });
                
                res.on('end', () => {
                    if (res.statusCode === 200) {
                        try {
                            const json = JSON.parse(data);
                            const txCount = json.transitions?.length || json.transactions?.length || 
                                          (Array.isArray(json) ? json.length : 0);
                            
                            if (txCount > 0) {
                                console.log(`✅ Found ${txCount} transactions`);
                                resolve({ success: true, count: txCount });
                            } else {
                                console.log(`⚠️  No transactions found (might be normal for new contracts)`);
                                resolve({ success: true, count: 0 });
                            }
                        } catch (e) {
                            console.log(`❌ Invalid JSON`);
                            resolve({ success: false });
                        }
                    } else {
                        console.log(`❌ HTTP ${res.statusCode}`);
                        resolve({ success: false });
                    }
                });
            }).on('error', (err) => {
                console.log(`❌ Error: ${err.message}`);
                resolve({ success: false });
            });
        });
        
        if (result.success) {
            return result;
        }
    }
    
    return { success: false };
}

async function runTests() {
    console.log('=== Aleo API Connection Test ===\n');
    console.log('Testing connection to Aleo blockchain APIs...');
    console.log(`Contracts to test: ${contracts.join(', ')}`);
    
    const results = [];
    
    // Test each endpoint with each contract
    for (const endpoint of endpoints) {
        console.log(`\n--- Testing endpoint: ${endpoint} ---`);
        
        for (const contract of contracts) {
            const result = await testEndpoint(endpoint, contract);
            results.push(result);
            
            if (result.success) {
                // Also test transaction fetching
                await testTransactions(endpoint, contract);
            }
        }
    }
    
    // Summary
    console.log('\n=== Summary ===');
    const successful = results.filter(r => r.success);
    console.log(`\nTotal tests: ${results.length}`);
    console.log(`Successful: ${successful.length}`);
    console.log(`Failed: ${results.length - successful.length}`);
    
    if (successful.length > 0) {
        console.log('\n✅ At least one endpoint is working!');
        console.log('The dashboard should be able to fetch live data.');
        
        // Find best endpoint
        const endpointSuccess = {};
        successful.forEach(r => {
            endpointSuccess[r.endpoint] = (endpointSuccess[r.endpoint] || 0) + 1;
        });
        
        const bestEndpoint = Object.entries(endpointSuccess)
            .sort((a, b) => b[1] - a[1])[0];
        
        console.log(`\nRecommended endpoint: ${bestEndpoint[0]} (${bestEndpoint[1]}/3 contracts working)`);
    } else {
        console.log('\n❌ All tests failed. Check your internet connection or try again later.');
    }
}

// Run the tests
runTests().catch(console.error);

File: tools/monitoring/test_aleo_connection.py
#!/usr/bin/env python3
"""Test Aleo blockchain API connection for our deployed contracts."""

import json
<<<<<<< HEAD
import urllib.error
import urllib.request
from datetime import datetime

CONTRACTS = [
    "hallucination_verifier.aleo",
    "agent_registry_v2.aleo",
    "trust_verifier_v2.aleo",
]

ENDPOINTS = [
    "https://api.explorer.provable.com/v1",
    "https://api.explorer.aleo.org/v1",
]


=======
import urllib.request
import urllib.error
from datetime import datetime

CONTRACTS = [
    'hallucination_verifier.aleo',
    'agent_registry_v2.aleo',
    'trust_verifier_v2.aleo'
]

ENDPOINTS = [
    'https://api.explorer.provable.com/v1',
    'https://api.explorer.aleo.org/v1',
]

>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
def test_endpoint(endpoint, contract_id):
    """Test if we can fetch contract data from an endpoint."""
    url = f"{endpoint}/testnet3/program/{contract_id}"
    print(f"\nTesting: {url}")
<<<<<<< HEAD

    try:
        req = urllib.request.Request(
            url,
            headers={
                "Accept": "application/json",
                "User-Agent": "Lamassu-Labs-Test/1.0",
            },
        )

=======
    
    try:
        req = urllib.request.Request(url, headers={
            'Accept': 'application/json',
            'User-Agent': 'Lamassu-Labs-Test/1.0'
        })
        
>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
        with urllib.request.urlopen(req, timeout=10) as response:
            data = json.loads(response.read().decode())
            print(f"✅ Success! Found program: {contract_id}")
            print(f"   Network: {data.get('network', 'unknown')}")
<<<<<<< HEAD
            if "functions" in data:
                print(f"   Functions: {len(data['functions'])}")
            if "mappings" in data:
                print(f"   Mappings: {len(data['mappings'])}")
            return True, data

=======
            if 'functions' in data:
                print(f"   Functions: {len(data['functions'])}")
            if 'mappings' in data:
                print(f"   Mappings: {len(data['mappings'])}")
            return True, data
            
>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
    except urllib.error.HTTPError as e:
        print(f"❌ Failed: HTTP {e.code}")
        return False, None
    except Exception as e:
        print(f"❌ Failed: {str(e)}")
        return False, None

<<<<<<< HEAD

=======
>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
def test_transactions(endpoint, contract_id):
    """Test if we can fetch transaction data."""
    paths = [
        f"/testnet3/program/{contract_id}/transitions?limit=5",
        f"/testnet3/transitions?program={contract_id}&limit=5",
    ]
<<<<<<< HEAD

    for path in paths:
        url = endpoint + path
        print(f"\nTesting transactions: {url}")

        try:
            req = urllib.request.Request(
                url,
                headers={
                    "Accept": "application/json",
                    "User-Agent": "Lamassu-Labs-Test/1.0",
                },
            )

            with urllib.request.urlopen(req, timeout=10) as response:
                data = json.loads(response.read().decode())

                # Count transactions
                tx_count = 0
                if "transitions" in data:
                    tx_count = len(data["transitions"])
                elif "transactions" in data:
                    tx_count = len(data["transactions"])
                elif isinstance(data, list):
                    tx_count = len(data)

=======
    
    for path in paths:
        url = endpoint + path
        print(f"\nTesting transactions: {url}")
        
        try:
            req = urllib.request.Request(url, headers={
                'Accept': 'application/json',
                'User-Agent': 'Lamassu-Labs-Test/1.0'
            })
            
            with urllib.request.urlopen(req, timeout=10) as response:
                data = json.loads(response.read().decode())
                
                # Count transactions
                tx_count = 0
                if 'transitions' in data:
                    tx_count = len(data['transitions'])
                elif 'transactions' in data:
                    tx_count = len(data['transactions'])
                elif isinstance(data, list):
                    tx_count = len(data)
                
>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
                if tx_count > 0:
                    print(f"✅ Found {tx_count} transactions")
                    # Show first transaction
                    first_tx = None
<<<<<<< HEAD
                    if "transitions" in data and data["transitions"]:
                        first_tx = data["transitions"][0]
                    elif "transactions" in data and data["transactions"]:
                        first_tx = data["transactions"][0]
                    elif isinstance(data, list) and data:
                        first_tx = data[0]

                    if first_tx:
                        print(f"   Latest: {first_tx.get('id', 'unknown')[:20]}...")
                        if "timestamp" in first_tx:
                            print(f"   Time: {first_tx['timestamp']}")
                else:
                    print("⚠️  No transactions found (might be normal)")

                return True

        except Exception as e:
            print(f"   Failed: {str(e)}")
            continue

    return False


=======
                    if 'transitions' in data and data['transitions']:
                        first_tx = data['transitions'][0]
                    elif 'transactions' in data and data['transactions']:
                        first_tx = data['transactions'][0]
                    elif isinstance(data, list) and data:
                        first_tx = data[0]
                    
                    if first_tx:
                        print(f"   Latest: {first_tx.get('id', 'unknown')[:20]}...")
                        if 'timestamp' in first_tx:
                            print(f"   Time: {first_tx['timestamp']}")
                else:
                    print("⚠️  No transactions found (might be normal)")
                
                return True
                
        except Exception as e:
            print(f"   Failed: {str(e)}")
            continue
    
    return False

>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
def main():
    """Run all connection tests."""
    print("=== Aleo API Connection Test ===")
    print(f"Testing at: {datetime.now()}")
    print(f"Contracts: {', '.join(CONTRACTS)}")
<<<<<<< HEAD

    working_endpoints = []

    for endpoint in ENDPOINTS:
        print(f"\n{'='*60}")
        print(f"Testing endpoint: {endpoint}")
        print("=" * 60)

        endpoint_works = False

        for contract in CONTRACTS:
            success, data = test_endpoint(endpoint, contract)

=======
    
    working_endpoints = []
    
    for endpoint in ENDPOINTS:
        print(f"\n{'='*60}")
        print(f"Testing endpoint: {endpoint}")
        print('='*60)
        
        endpoint_works = False
        
        for contract in CONTRACTS:
            success, data = test_endpoint(endpoint, contract)
            
>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
            if success:
                endpoint_works = True
                # Also test transactions
                test_transactions(endpoint, contract)
<<<<<<< HEAD

        if endpoint_works:
            working_endpoints.append(endpoint)

    # Summary
    print(f"\n{'='*60}")
    print("SUMMARY")
    print("=" * 60)

=======
        
        if endpoint_works:
            working_endpoints.append(endpoint)
    
    # Summary
    print(f"\n{'='*60}")
    print("SUMMARY")
    print('='*60)
    
>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
    if working_endpoints:
        print(f"\n✅ SUCCESS! Found {len(working_endpoints)} working endpoint(s):")
        for ep in working_endpoints:
            print(f"   - {ep}")
        print("\nThe live dashboard should work with these endpoints.")
        print("\nTo view the dashboard:")
        print("1. Open: tools/monitoring/dashboard-live.html")
        print("2. Or run: python3 -m http.server 8000")
        print("   Then visit: http://localhost:8000/dashboard-live.html")
    else:
        print("\n❌ All connection attempts failed.")
        print("Please check your internet connection and try again.")

<<<<<<< HEAD

if __name__ == "__main__":
    main()
=======
if __name__ == "__main__":
    main()
>>>>>>> 175afbc51eef8fe475bbc42703bff3cf5a864752
